codalab/model/bundle_model.py

```diff
@@ -677,6 +677,58 @@ def get_bundle_uuids(self, conditions, max_results):
 
         return self._execute_query(query)
 
+    def get_memoized_bundles(self, user_id, command, dependencies):
+        # Select bundles that has the given command from the bundle table
+        filter_on_command = (
+            select([cl_bundle.c.uuid])
+            .select_from(cl_bundle)
+            .where(and_(cl_bundle.c.command == command, cl_bundle.c.owner_id == user_id))
+            .alias("filter_on_command")
+        )
+        # Get a list of uuids that matches with the given dependencies from the cl_bundle table.
+        # This step is to standardize uuid for future comparison.
+        dep_clause = [
+            cl_bundle_dependency.c.parent_uuid.like('%' + uuid + '%') for uuid in dependencies
```

Comment:
Why is this not exact equality?

Reply:
The `uuid` passed in from the command line might be a partial string. Then a string pattern matching operation could help us to find the exact bundle.

Reply:
When does this happen?  `uuid` should really always mean the full uuid.

Reply:
This could happen in cases like this `cl run :0x23456 ...`. Then the `uuid` that gets passed into the function will be a partial one.

Reply:
I think in that case, the partial uuid should get resolved to a full uuid?

Reply:
Yes, that's what we are trying to do here. Basically, I was trying to resolve the partial uuid to a full uuid so that we can do a comparison later on.

Reply:
I thought the canonicalization happens earlier.  Are you sure at this point `uuid`s aren't just full uuids?

Reply:
Yes, you are right. I have updated this part with the object that has resolved the partial uuid issue.